### ISSUE TITLE
Hoisted dependencies possibly getting flagged as dev only

### ISSUE THREAD

**Issue by jostrander** [MarshallOfSound/flora-colossus#7]
Minimal repro with package.json:
```
{
  "optionalDependencies": {
    "dbus-next": "0.6.1",
    "mpris-service": "2.1.0",
    "xml2js": "^0.4.19"
  },
  "devDependencies": {
    "electron-packager": "^13.1.1",
    "galactus": "^0.2.1",
    "jimp": "^0.2.27",
    "plist": "^2.0.1"
  }
}
```
`yarn`

then run: 

```
DEBUG=flora-colossus node -e "new (require('flora-colossus').Walker)('.').walkTree().then(e => e.map(x => x.path.includes('xml') ? console.log(x) : false ) )"
 ```

xmlbuilder is only showing as a `dev` dependency, even though `yarn why` shows as required by deps in optionaldeps:
```
yarn why xmlbuilder
yarn why v1.16.0
warning package.json: No license field
[1/4] Why do we have the module "xmlbuilder"...?
[2/4] Initialising dependency graph...
warning No license field
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "xmlbuilder@9.0.7"
info Has been hoisted to "xmlbuilder"
info Reasons this module exists
   - Hoisted from "xml2js#xmlbuilder"
   - Hoisted from "electron-packager#plist#xmlbuilder"
   - Hoisted from "electron-packager#electron-osx-sign#plist#xmlbuilder"
info Disk size without dependencies: "172KB"
info Disk size with unique dependencies: "172KB"
info Disk size with transitive dependencies: "172KB"
info Number of shared dependencies: 0
=> Found "plist#xmlbuilder@8.2.2"
info This module exists because "plist" depends on it.
info Disk size without dependencies: "172KB"
info Disk size with unique dependencies: "172KB"
info Disk size with transitive dependencies: "172KB"
info Number of shared dependencies: 0
Done in 0.18s.
```

**Comment by MarshallOfSound** [comment]
Hm, will investigate, glancing at our code and tests this case should be handled and the package should be considered optional.

https://github.com/MarshallOfSound/flora-colossus/blob/master/src/depTypes.ts#L80-L81

**Comment by jostrander** [comment]
I left out the actual results from the walk, here's what that produces:
```js
{ depType: 1,
  path: '/home/jostrander/data/Code/other/test-destroyer/node_modules/xmlbuilder',
  name: 'xmlbuilder' }
{ depType: 1,
  path: '/home/jostrander/data/Code/other/test-destroyer/node_modules/xmldom',
  name: 'xmldom' }
{ depType: 1,
  path: '/home/jostrander/data/Code/other/test-destroyer/node_modules/parse-bmfont-xml',
  name: 'parse-bmfont-xml' }
{ depType: 1,
  path: '/home/jostrander/data/Code/other/test-destroyer/node_modules/xml-parse-from-string',
  name: 'xml-parse-from-string' }
{ depType: 2,
  path: '/home/jostrander/data/Code/other/test-destroyer/node_modules/xml2js',
  name: 'xml2js' }
{ depType: 1,
  path: '/home/jostrander/data/Code/other/test-destroyer/node_modules/plist/node_modules/xmlbuilder',
  name: 'xmlbuilder' }
```
It definitely seems like an edge case, as it seemed to work fine until I had enough deps to get yarn to hoist it, though I haven't confirmed that's what caused it.

**Comment by MarshallOfSound** [comment]
@jostrander Fixed in `1.0.1` 👍

**Comment by jostrander** [comment]
This is going to need to be bumped all the way up to electron-packager, should I open issues for all the intermediary repos?
